### PR TITLE
App data handling balance api

### LINKED ISSUE #429: App data handling - Balance API

### PR DESCRIPTION
**Pull request by vishnukvnvv**
Closes #429

### CODE REVIEW

File: bundler/src/contracts/usdc_provider.rs [hunk 1]
```diff
@@ -44,4 +45,12 @@ impl USDCProvider {
 
         Ok(data.unwrap())
     }
+
+    pub async fn balance_of(client: &Web3Client, address: Address) -> Result<U256, String> {
+        client
+            .get_usdc_provider()
+            .balance_of(address)
+            .await
+            .map_err(|_| String::from("Failed to get balance"))
```
Comment (ska1296) [inline bundler/src/contracts/usdc_provider.rs:54]: please log the error you are getting from the client

File: bundler/src/handlers/wallet.rs [hunk 2]
```diff
@@ -28,19 +28,21 @@ pub async fn get_address(
...
-    respond_json(data)
+    let data = BalanceService::get_wallet_balance(
+        pool.get_ref(),
+        provider.get_ref(),
+        &balance_request.get_chain(),
+        &balance_request.get_currency(),
+        user.into_inner(),
+    )
+    .await?;
+    toad_response(BaseResponse::init(data))
```
Comment (ska1296) [inline bundler/src/handlers/wallet.rs:45]: don't introduce `toad_response`.

File: bundler/src/server.rs [hunk 3]
```diff
@@ -176,10 +171,10 @@ pub async fn run(service: ToadService, server: Server) -> std::io::Result<()> {
             .configure(routes)
             .app_data(Data::new(service.hello_world_service.clone()))
             .app_data(Data::new(service.wallet_service.clone()))
-            .app_data(Data::new(service.balance_service.clone()))
             .app_data(Data::new(service.transfer_service.clone()))
             .app_data(Data::new(service.admin_service.clone()))
             .app_data(Data::new(service.token_metadata_service.clone()))
+            .app_data(Data::new(service.web3_client.clone()))
```
Comment (ska1296) [inline bundler/src/server.rs:177]: don't think you need this clone?

**Review by ska1296 — CHANGES_REQUESTED** [review]

File: bundler/src/server.rs (hunk 3 above)
Comment (vishnukvnvv) [reply bundler/src/server.rs:177]: Without clone, compilation fails with error

**Review by ska1296 — APPROVED** [review]

**Pull request merged**